dom/base/Element.cpp

```diff
@@ -3903,10 +3920,15 @@ void Element::SetInnerHTML(const nsAString& aInnerHTML,
                            nsIPrincipal* aSubjectPrincipal,
                            ErrorResult& aError) {
 
-  // TaintFox: innerHTML sink.
-  ReportTaintSink(aInnerHTML, "innerHTML", this);
+  // TaintFox: innerHTML sink - don't set for template elements
+  if (!IsTemplateElement()) {
+    ReportTaintSink(aInnerHTML, "innerHTML", this);
```

Comment:
Check whether this makes sense - effectively we will no longer report sinks for innerHTML template elements. The motivation is that scripts are not executed for templates.